root/static/scripts/edit/externalLinks.js

```diff
@@ -1460,7 +1454,28 @@ export class ExternalLink extends React.Component<LinkProps> {
         if (isBlog) {
           faviconClass = 'blog';
         } else {
-          faviconClass = 'no';
+          const isReview = props.relationships.some(link => {
+            const linkType = link.type
+              ? linkedEntities.link_type[link.type]
+              : null;
+            if (linkType) {
+              return /^review$/.test(linkType.name);
+            }
+            return false;
+          });
+          if (isReview) {
+            faviconClass = 'review';
+          } else {
+            for (const key of Object.keys(FAVICON_CLASSES)) {
```

Comment:
This whole `render` method is kinda repetitious now with three separate calls to `props.relationships.some`. Would it be possible to replace all of the code here with a single loop, and perform the relevant checks inside the loop (IIUC we can just `break` after the first match)?

Reply:
> Well, if the same URL is used for both blog and official site for example we want to show the official site icon, it seems.

Did we actually intend to support that? I assume that'd only matter for labels, but I only found 14 with both (and several are dead, broken, or not blogs). 😆

Thanks for the improved code though, it's definitely clearer. In the commit message you mentioned performance, and the most significant way to improve that would be to remove the regular expression. I'd remove the `continue` statements because they hurt clarity and don't help performance, and would just assign directly to `faviconClass`.
```js
  let faviconClass = 'no';

  for (const relationship of relationships) {
    const linkType = relationship.type
      ? linkedEntities.link_type[relationship.type]
      : null;

    if (linkType) {
      // If we find a homepage, that's the icon we want and we're done
      if (
        linkType.name === 'official homepage' ||
        linkType.name === 'official site'
      ) {
        return 'home';
      } else if (linkType.name === 'blog') {
        faviconClass = 'blog';
      } else if (linkType.name === 'review') {
        faviconClass = 'review';
      }
    }
  }

  if (faviconClass !== 'no') {
    return faviconClass;
  }
```

Reply:
The `continue` statement for blog at least was there to ensure we don't overwrite a previously found "blog" with "review" since blog has priority right now. It might be it doesn't matter at all which of the two has the priority, but I didn't want to change the conceptual way of how we do this, just clarify/simplify.

Reply:
After a discussion we decided that most if any cases where the same thing is blog and review should be bad data so it doesn't matter really which icon they get, so will use the suggested code.